lib/TPP/Transforms/Utils/VNNIUtils.cpp

```diff
@@ -20,22 +21,54 @@ namespace mlir {
 namespace vnni {
 namespace utils {
 
-std::optional<int64_t> getVnniBlockingFactor(Type type) {
+std::optional<int64_t> getVnniBlockingFactor(Type type, Operation *op) {
   auto elementType = getElementTypeOrSelf(type);
-  if (elementType.isBF16())
+  if (elementType.isBF16()) {
+    // Check if a VNNI factor hint is associated to the IR via DLTI.
+    auto deriveVnniFromDLTI = [&]() -> std::optional<int64_t> {
```

Comment:
I would extract this in to a proper method, for some future DLTI helper.

Reply:
Added helper.
@rolfmorel I think such query on strings could also be upstreamed.